compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -1216,7 +1217,7 @@ pub fn linker_and_flavor(sess: &Session) -> (PathBuf, LinkerFlavor) {
 
     if let Some(ret) = infer_from(
         sess,
-        sess.target.linker.clone().map(PathBuf::from),
+        sess.target.linker.as_ref().map(|l| PathBuf::from(l.as_ref())),
```

Comment:
Are 2 `as_refs` necessary here?

Reply:
Seems like yes and no, the first one is responsible for `Option<Cow<'static, str>>` to `Option<&Cow<'static, str>>` and the second one for `&Cow<'static, str>` to `&str>` but it can be replaced by `as_deref()` with does a similar thing: `Option<Cow<'static, str>>` to `&Option<&str>`.